README.md

```diff
@@ -76,6 +76,94 @@ var mocha = new Mocha({
 })
 ```
 
+### Append properties to testcase
+
+You can provide metadata for each testcase, i.e., properties for each `testcase`.
+Some tools (e.g., Xray) may take advantage of this to provide additional information related to the execution of each testcase or to perform other operations while importing test results (e.g., linking results to existing test cases or stories on some tool, such as Jira).
+
+Zero, one or more properties may be provided using this metadata. You can use whatever name and attributes for the properties.
+
+Properties can be defined by:
+
+- a name (string) and an inline value (string)
+- a name (string) and a content (multiline string)
+- or a name (string), and a set of objects having one or more attributes, and optionally content
+
+```xml
+<testsuites>
+  <testsuite>
+    <testcase name="should do some stuff" time="1.7390" classname="demo">
+      <properties>
```

Comment:
@bitcoder i just noticed that this is using a `properties` element within a `testcase` which is not valid for Jenkins: https://github.com/jenkinsci/xunit-plugin/blob/master/src/main/resources/org/jenkinsci/plugins/xunit/types/model/xsd/junit-10.xsd

Jenkins only allows `property` elements under a `testsuite` element. I could be wrong about this, but we could prove it with a test case that runs this output against the junit XSD. There are a couple of tests in the spec file that will show you how to do this.

Reply:
Well, `properties` are usually an element that appear under `testsuite`, that's correct. junit has not an official XSD though AFAIK. I had checked before with the Junit project leader Marc Philipp , about the feasibility of having also `properties`under the `testcase` element and we didn't see a problem with it; however, some tools may have a more strict validation and support only what ant provided back then, on the first releases of the Junit report format.
The current code **only** adds properties if testCaseMetadata is provided, which ensures we keep backward compatible; this is independent of the "ant/jenkins" mode flags. However, I understand if you want to have to inhibit this feature if those modes are active.
Having said that, shall we keep the current logic or do you prefer to change it to only make it applicable if not in Jenkins mode and also not in ant mode?

Reply:
Just to complement, Datadog [also supports](https://docs.datadoghq.com/continuous_integration/setup_tests/junit_upload/) properties at testcase level, which 
will also be supported by this PR.